os/env/path-extender-posix/path-extender-posix.ts

```diff
@@ -72,7 +72,13 @@ async function updateShell (
 }
 
 async function setupShell (shell: 'bash' | 'zsh', dir: string, opts: AddDirToPosixEnvPathOpts): Promise<PathExtenderPosixReport> {
-  const configFile = path.join(os.homedir(), `.${shell}rc`)
+  const shellConfigName = `.${shell}rc`
+  let configFile!: string
+  if (shell === 'zsh') {
+    configFile = path.join((process.env.ZDOTDIR || os.homedir()), shellConfigName)
+  } else {
+    configFile = path.join(os.homedir(), shellConfigName)
+  }
```

Comment:
create a new function for getting the configFile path

Reply:
done